superset/viz.py

```diff
@@ -1231,6 +1232,39 @@ def get_data(self, df):
         return df.to_dict(orient='records')
 
 
+class ChordViz(BaseViz):
+
+    """A Chord diagram"""
+
+    viz_type = "chord"
+    verbose_name = _("Directed Force Layout")
+    credits = '<a href="https://github.com/d3/d3-chord">Bostock</a>'
+    is_timeseries = False
+
+    def query_obj(self):
+        qry = super(ChordViz, self).query_obj()
+        fd = self.form_data
+        qry['groupby'] = [fd.get('groupby'), fd.get('columns')]
+        qry['metrics'] = [fd.get('metric')]
+        return qry
+
+    def get_data(self, df):
```

Comment:
`get_data` receives a pandas dataframe (df) that correspond to what is returned when applying your query object defined above. The shape of that dataframe is related to what you ask for in that `query_obj`.

Now whatever `get_data` returns needs to be json-serializable, and will be made available to the javascript visualization function as `payload.data`.